extensions/vscode/webviews/homeView/src/components/views/secrets/Secret.vue

```diff
@@ -0,0 +1,94 @@
+<template>
+  <form v-if="showInput" @submit.prevent="updateSecret">
+    <input
+      ref="input"
+      class="secret-input"
+      v-model="inputValue"
+      aria-label="Type secret value. Press Enter to confirm or Escape to cancel."
+      @keydown.escape="showInput = false"
+    />
+  </form>
+  <TreeItem
+    v-else
+    :title="name"
+    :description="secretValue ? '••••••' : undefined"
+    :actions="actions"
+    codicon="codicon-lock-small"
+    :tooltip="tooltip"
+    align-icon-with-twisty
+    @click="inputSecret"
+  />
+</template>
+
+<script setup lang="ts">
+import { computed, ref, nextTick } from "vue";
+
+import TreeItem from "src/components/tree/TreeItem.vue";
+import { useHomeStore } from "src/stores/home";
+import { ActionButton } from "src/components/ActionToolbar.vue";
+
+interface Props {
+  name: string;
```

Comment:
I would have expected that this component would be a UI component that would accept its name and value through a prop and emit a change when the value changed. I would then have expected the view using this component to have to deal with the storage of the secret value. 

While this approach will work, it makes this component extremely dedicated to this particular input vs. having a component that allows its value to be edited. I think we would benefit from having it be a bit more general purpose, especially when you think of using it for environment variables as well.

Reply:
I actually started writing it this way at the start, but found it more difficult to keep the `home.secrets` Map in line with the value that was being `v-model`ed in `Secret.vue` instead I made it just reflect the state of `home.secrets.get('name')` for the `name` key provided

Reply:
The main difficulty when using `v-model` was grabbing the most recent value when swapping over to the input mode. It worked great when setting the value, but not when the value changed out from under the component.

Reply:
Did you get a chance to play with the approach we discussed via zoom? That would be where outside component would pass in the map value from the store and then update the store when receiving the change emitted event from the component?

Reply:
I did mess around with this, and it ended up re-creating the Map in the `Secrets` component a bit as an array so each `v-model` accessed an index in that array. It felt duplicative to me.